tdash/functions.py

```diff
@@ -95,4 +95,25 @@ def result(fn_or_value, *args, **kwargs):
     """
     if inspect.isfunction(fn_or_value):
         return safe_partial(fn_or_value, *args, **kwargs)()
-    return fn_or_value
\ No newline at end of file
+    return fn_or_value
+
+
+###
+# I forgot what the note you wanted me to add here was, something about tuples as keys?
+###
+def memoize(f):
+    """
+    Decorator to memoize a function taking one or more arguments.
+    Uses per-instance cache. Does not have a method to reset cache.
+
+    :param decorated_function: Function to be decorated
+    : return:
+    """
+    cache = {}
+    def decorated_function(*args):
```

Comment:
Does this work with keyword arguments (do they get passed through? Do they get cached?)